guides/introduction/Embedded Schemas.md

```diff
@@ -92,57 +87,124 @@ end
 
 ```elixir
 defmodule User do
-  use Ecto.Schema
-
   # ...
 
   def changeset(user, attrs \\ %{}) do
+    user
+    |> cast(attrs, [:full_name, :email, :avatar_url])
+    |> cast_embed(:profile, required: true)
+  end
+end
+
+changeset = User.changeset(%User{}, %{profile: %{online: true}})
+changeset.valid? # => false; "visibility can't be blank"
+changeset = User.changeset(%User{}, %{profile: %{online: true, visibility: :public}})
+changeset.valid? # => true
+```
+
+In situations where you have kept the embedded schema within the parent module, e.g., you have not extracted a UserProfile, you can still have custom changeset functions for the embedded data within the parent schema.
+
+```elixir
+defmodule User do
+  use Ecto.Schema
+
+  schema "users" do
+    field :full_name, :string
+    field :email, :string
+    field :avatar_url, :string
+    field :confirmed_at, :naive_datetime
+
+    embeds_one :profile, Profile do
+      field :online, :boolean
+      field :dark_mode, :boolean
+      field :visibility, Ecto.Enum, values: [:public, :private, :friends_only]
+    end
+
+    timestamps()
+  end
+
+  def changeset(%User{} = user, attrs \\ %{}) do
     user
     |> cast(attrs, [:full_name, :email])
-    # By default it calls UserProfile.changeset/2, pass the :with option to change it
-    |> cast_embed(:user_profile, required: true)
+    |> cast_embed(:profile, required: true, with: &profile_changeset/2)
+  end
+
+  def profile_changeset(profile, attrs \\ %{}) do
+    profile
+    |> cast(attrs, [:online, :dark_mode, :visibility])
+    |> validate_required([:online, :visibility])
   end
 end
+
+changeset = User.changeset(%User{}, %{profile: %{online: true, visibility: :public}})
+changeset.valid? # => true
 ```
+
+### Querying embedded data
+
+Once you have written embedded data to the database, you can use it in queries on the parent schema.
+
+```elixir
+user_changeset = User.changeset(%User{}, %{profile: %{online: true, visibility: :public}})
+{:ok, _user} = Repo.insert(user_changeset)
+
+(Ecto.Query.from u in User, select: {u.profile["online"], u.profile["visibility"]}) |> Repo.one
+# => {true, "public"}
+
+(Ecto.Query.from u in User, select: u.profile, where: u.profile["visibility"] == :public) |> Repo.all
+# => [
+#  %UserProfile{
+#    id: "...",
+#    online: true,
+#    dark_mode: nil,
+#    visibility: :public
+#  }
+#]
+```
+
+In typical databases where `:map`s are mapped to JSONB (like Postgres), Ecto is constructing the appropriate jsonpath queries for you. To see more examples of valid syntax for embedded schemas, see [`json_extract_path/2`](https://hexdocs.pm/ecto/Ecto.Query.API.html#json_extract_path/2).
```

Comment:
> In databases where `:map`s are stored as JSONB (like Postgres)

Reply:
Reworked the paragraph and incorporated the feedback, <3